python/hawkey/sack-py.cpp

```diff
@@ -52,6 +52,22 @@ typedef struct {
     PyObject *custom_package_class;
     PyObject *custom_package_val;
     PyObject * ModulePackageContainerPy;
+
+    // g_log handlers: a pair of <handler_id, log_domain>
+    // Multiple sacks can be created during a run of an application and each
+    // sack opens a log file and registers two g_log handlers. To avoid dangling
+    // handlers with invalid FILE pointers (we close them when destroying the
+    // sack), we need to keep track of the handlers so that we can also remove
+    // them. g_log_remove_handler() needs both the id and the domain, hence the
+    // vector of pairs.
+    //
+    // g_log is clever about adding log handlers. It does store all handlers
+    // registered for a given domain, but only the one that was registered last
+    // is used. If you remove the last registered one, the next in line will be
+    // used. That means stacking sacks is ok, the handler from the last
+    // undeleted sack will be the one that is used.
+    std::vector<std::pair<guint, std::string>> log_handlers;
```

Comment:
This is not good. Python/C API has some rules (https://docs.python.org/3/c-api/typeobj.html).
Look into `sack_new`. Memory for `_SackObject` is allocated by `tp_alloc`. The `p_alloc` allocates memory. It knows nothing about constructor of C++ members. Clean solution is that member is a pointer to C++ object. And the pointer is initialized using `new` in `sack_new` and destroyed using `delete` in `sack_dealloc`.